libnmstate/netapplier.py

```diff
@@ -332,3 +333,11 @@ def _cleanup_iface_ethernet_state_sanitize(desired_state, current_state):
                 ethernet_current_state.pop(key, None)
         if not ethernet_desired_state:
             desired_state.pop('ethernet', None)
+
+
+def _cleanup_iface_ipv6_link_local_addr(desired_state, current_state):
+    for state in (desired_state, current_state):
+        for addr_dict in state.get('ipv6', {}).get('address', []):
```

Comment:
I may miss something, but I think addr_dict is actually a list here. Please make sure you cover this flow here using tests.
BTW: Please do not use the name of the structure in the name. Use something like addr_data or address_state

Reply:
The `state['ipv6']['address'] a list of dict. So `addr_dict` is a dict here. I will rename it of course.
```json
"address": [
    {
        "ip": "fe80::8097:51a0:9ff4:c7a1",
        "prefix-length": 64
    }
],
```